API.md

```diff
@@ -114,3 +115,20 @@ npmlock2nix.build {
   };
 }
 ```
+
+### Source modifications
+
+`node_modules` takes a `sourceMods` argument, which allows you to modify sources of individual npm packages you depend on, useful for adding Nix-specific fixes to packages. This could be used for patching interpreter or paths, or to replace vendored binaries with ones provided by Nix:
+
+```nix
+npmlock2nix.node_modules {
+  sourceMods = {
+    # <package name> = { version }: <derivation attributes>;
+
+    # Patching shebangs
+    node-pre-gyp = _: {
+      postPatch = "patchShebangs bin";
```

Comment:
The interface here is a very leaky abstraction. We just plug some additional variables into the `runCommand` but only some will ever actually have an impact. I'd rather prefer an interface that isn't passing everything through from the user but defines a stricter interface.

The alternative is to provide an `overrideAttrs` interface here instead.

I prefer the first approach as it comes with a more limited set of features that we have to take care of.

Reply:
Sounds good. I think we can limit this to only `patches` and `postPatch` for now, but it could be extended with support for other attributes in the future (for now other attributes should give an error)

Reply:
After another thought, I think it's probably better to just support all attributes, both to keep it simpler, but also because people might have different needs. Notably there's also `prePatch`, you can add dependencies in `buildInputs` as needed by `patchShebangs`, you may want to set `NIX_DEBUG` for debugging, you may want to set `sourceRoot` to only get a subdirectory, you may want to set `dontUnpack` to preserve the original `.tgz` and patch that with `tar` directly, you may want to set environment variables.

I pushed support for this in this PR now